diff --git a/index.js b/index.js
index bd438442ea43e7be342d1604f661d72db51706a7..ed52b7077250de0999b824b4fd049630db50336f 100644
--- a/index.js
+++ b/index.js
@@ -70,7 +70,15 @@ function run(tscPath) {
         }
     }
 }
-function resolveTscPath(tscPath = require.resolve('typescript/lib/tsc')) {
+function resolveTscPath(tscPath) {
+    if (!tscPath) {
+        try {
+            tscPath = require.resolve('typescript/lib/tsc');
+        } catch {
+            const tsPkg = require.resolve('typescript/package.json');
+            tscPath = path.join(path.dirname(tsPkg), 'lib', 'tsc.js');
+        }
+    }
     try {
         const { name } = require(path.join(tscPath, '..', '..', 'package.json'));
         if (name === '@typescript/typescript6') {
